packages/app/src/cli/models/app/loader.ts

```diff
@@ -788,6 +791,38 @@ class AppLoader<TConfig extends AppConfiguration, TModuleSpec extends ExtensionS
         })
       })
   }
+
+  /**
+   * Determines the appropriate dev application URLs for the current configuration.
+   * If the app proxy configuration has changed, preserves the tunnel URL but regenerates
+   * config-dependent parts. Otherwise, uses the cached URLs.
+   */
+  private getDevApplicationURLs(currentConfiguration: TConfig): ApplicationURLs | undefined {
+    const previousDevUrls = this.previousApp?.devApplicationURLs
+
+    if (!previousDevUrls) {
+      // No previous URLs to preserve, will be generated later
+      return undefined
+    }
+
```

Comment:
We can simplify this and force the devApplicationURLs everytime:

```ts
const previousDevUrls = this.previousApp?.devApplicationURLs
if (!previousDevUrls) return undefined

// get the `authCallbackPath` here somehow

return generateApplicationURLs(previousDevUrls.applicationUrl, authCallbackPath, currentAppConfig.app_proxy)
```

Reply:
You'll need to inject the `webs` config into this method (you have it available on line 376)